app/src/main/java/com/client/fire_and_water/CheckInFragment.kt

```diff
@@ -0,0 +1,39 @@
+package com.client.fire_and_water
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.fragment.findNavController
+import com.client.fire_and_water.databinding.FragmentCheckInBinding
+
+class CheckInFragment : Fragment() {
+    private var _binding: FragmentCheckInBinding? = null
+
+    // This property is only valid between onCreateView and
+    // onDestroyView.
+    private val binding get() = _binding!!
```

Comment:
Couldn't there be NPE here?

Reply:
Weell it can be, i guess? if it's called before onCreateView or after onDestroyView
but it's private..

..and it came from default app

Reply:
If there is a call to `binding`, prior to initialization, then there will be an NPE, whether private or not. Look what `!!` means. Here the question is just in the order of method and property calls.